script.js

```diff
@@ -0,0 +1,138 @@
+let currencyRates = {
+    timestamp: 1519296206,
+    base: "EUR",
+    date: "2021-03-17",
+    rates: {
+        USD: 1.23396
+    }
+};
+
+function insertCurrencyRate() {
+    const base = document.getElementById('base').value.toUpperCase();
```

Comment:
Using `toUpperCase()` is an excellent idea. You should `trim` the value too so you can avoid situations like the following where you have two instances of `CHN`
```js
{
  "timestamp": 1720545130050,
  "base": "EUR",
  "date": "2024-07-09",
  "rates": {
    "USD": 1.23396,
    "CHN": 142,
    "     CHN  ": 321
  }
}
```